includes/class-amp-theme-support.php

```diff
@@ -1036,10 +1055,36 @@ public static function prepare_response( $response, $args = array() ) {
 				'allow_dirty_styles'      => self::is_customize_preview_iframe(), // Dirty styles only needed when editing (e.g. for edit shortcodes).
 				'allow_dirty_scripts'     => is_customize_preview(), // Scripts are always needed to inject changeset UUID.
 				'disable_invalid_removal' => $is_validation_debug_mode,
+				'enable_response_caching' => ! defined( 'WP_DEBUG' ) || true !== WP_DEBUG || $is_validation_debug_mode,
```

Comment:
Instead of `$is_validation_debug_mode` I think `AMP_Validation_Utils::should_validate_response()` should be used instead. The debug mode is essentially verbose validation, and we wouldn't want caching to be done during validation.